terraform/custom_domains/infrastructure/terraform.tf

```diff
@@ -0,0 +1,22 @@
+terraform {
+
+  required_version = "= 1.3.5"
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "3.43.0"
```

Comment:
Can we use the same version as https://github.com/DFE-Digital/apply-for-qualified-teacher-status/blob/main/terraform/aks/terraform.tf#L9